### DISCUSSION TITLE
REST API - "Query Error: Only Absolute URLs are supported"

### DISCUSSION THREAD

**Discussion by cohenrobinson** [Budibase/budibase#2491] [Help] [answered]
Hi,

Just started using Budibase, so excuse my ignorance if I'm incorrectly identifying an issue, but I'm currently developing a REST API, and running this in a local test environment on the same PC that I'm using to run the Budibase instance.

## Expected Behaviour
That a REST API request can be requested from an absolute URL that's entered.

## Actual Behaviour
A REST API cannot be requested from an absolute URL, as the URL is _incorrectly_ identified as a relative URL.

If you'll see the attached screenshot, I've used a absolute URL (to my knowledge) but I keep receiving the response __"Query Error: Only Absolute URLs are supported"__.

<img width="500" alt="Screen Shot 2021-08-31 at 1 04 25 am" src="https://user-images.githubusercontent.com/39592873/131360743-a3af97ac-7641-4a73-b60e-f4317048dc15.png">

Note that I've just put a random url in there for the moment (this isn't the actual url I'm referencing) but the error is equivalent irrespective of the base-URL, or query, that's entered.

If I'm doing something wrong on my end, some assistance would be greatly appreciated. Thanks!

**Comment by mslourens** [comment]
It is possible you forgot to press the save button after changing the URL on the first REST configuration page. The preview uses the data in the database to fire the rest endpoint, but the UI displays the state of the app, which can be different. Does the URL change when you refresh the page? If so, please press the Save button before clicking the `Add query` button.

**Reply by cohenrobinson** [reply]
Thanks Maurits. When pressing `save` before the `add query` button, it will default back to `localhost` irrespective of what I've entered before pressing save.

**Comment by mjashanks** [comment, marked as answer]
Hi @cohenrobinson . I suspect that your issue is all related to issue that you mention as:

>  it will default back to `localhost` irrespective of what I've entered before pressing save

This is a bug that we will get fixed ASAP.

What I think is happening is that (underneath) it's it's trying to run against the URL `localhost/example`, which it sees as a relative path, due to it not having a scheme, e.g. `http://`

**Reply by cohenrobinson** [reply]
Fantastic. Thanks for your help, Michael. Looking forward to seeing it fixed so I can begin using Budibase!

**Reply by mjashanks** [reply]
No problem! It's actually fixed already, but not released. Should be released in the next few days.

**Comment by cryodream** [comment]
Serving simple rest api with [json-server - npm](https://www.npmjs.com/package/json-server#alternative-port)

### Working perfectly in chrome
![2021-09-14_07-53-31__chrome](https://user-images.githubusercontent.com/844036/133254196-368fbf17-1d90-4f70-9310-3282f0063ae0.png)

It also works over local network, when running json-server on my other machines.

### But localhost still does not work, which is a bummer 😕
![2021-09-14_07-51-35__chrome](https://user-images.githubusercontent.com/844036/133254139-bd9e733f-9e5e-4d6b-b7f0-0e20b6c8a338.png)
![2021-09-14_07-52-43__chrome](https://user-images.githubusercontent.com/844036/133254152-cfb3c8be-fbec-432a-a4c4-f385072c1047.png)

BTW, using `127.0.0.1` does not work also.

**Reply by mjashanks** [reply]
Hi there. Budibase is running inside a docker container, so `localhost` does not point to the host machine.

If you are using Mac or Windows, you should use `http://host.docker.internal:7451`. If using Linux, it'll be `http://172.17.0.1:7451`.

**Reply by cryodream** [reply]
Thank You @mjashanks 
Using `http://host.docker.internal:7451` works 👍
